src/MockFactory.ts

```diff
@@ -64,111 +65,146 @@ const executeOperation = (
 
 interface Options {
   mocks?: ResolverMap
+  addTypename?: Boolean
+}
+
+interface MockOptions<TVariables> {
+  variables?: TVariables
+  mocks?: ResolverMap
+  addTypename?: Boolean
 }
 
 class MockFactory {
   private schema: GraphQLSchema
   private mocks: ResolverMap
+  private addTypename: Boolean
 
   private mockSchema(options: { mocks?: ResolverMap } = {}) {
     const clonedSchema = transformSchema(this.schema, [])
-    const mergedMocks = mergeResolvers(this.mocks, options.mocks || {})
+    const mergedMocks = mergeResolvers(this.mocks, options.mocks ?? {})
     addMockFunctionsToSchema({ schema: clonedSchema, mocks: mergedMocks })
     return clonedSchema
   }
 
-  constructor(schema: IntrospectionQuery, { mocks }: Options) {
-    this.schema = generateMockSchema(schema)
-    this.mocks = mocks || {}
+  private maybeAddTypenameToDocument = (
+    document: DocumentNode,
+    overrideAddTypename: Boolean | undefined,
+  ) => {
+    const addTypename = overrideAddTypename ?? this.addTypename
+    // ? parse(print(addTypenameToDocument(document)))
+    return addTypename ? addTypenameToDocument(document) : document
+  }
+
+  constructor(
+    schema: IntrospectionQuery,
+    { mocks, addTypename = false }: Options = {},
+  ) {
+    this.schema = buildMockSchema(schema)
+    this.mocks = mocks ?? {}
+    this.addTypename = addTypename
   }
 
-  mockFragment = <TData = any>(
+  mockFragment = <TData = any, TVariables = any>(
     fragment: DocumentNode,
-    { mocks }: Options = { mocks: {} },
+    options: MockOptions<TVariables> = {},
   ): TData => {
-    const mainDefinition = getMainDefinition(fragment)
-    const fragmentWithTypename = addTypenameToDocument(fragment)
+    const { variables = {}, mocks = {}, addTypename } = options
+
+    const fragmentDocument = this.maybeAddTypenameToDocument(
+      fragment,
+      addTypename,
+    )
+    const mainDefinition = getMainDefinition(fragmentDocument)
 
     if (mainDefinition.kind !== 'FragmentDefinition') {
       throw new Error(
         'MockFactory: mockFragment only accepts fragment documents',
       )
     }
 
-    /* eslint-disable graphql/template-strings */
     const typeName = mainDefinition.typeCondition.name.value
     const fieldName = `mock__${typeName}`
-    const query = gql`
+    const query = parse(/* GraphQL */ `
```

Comment:
Why this change? You no like gql tag? You like print?

Reply:
gql tag was complaining that I was creating a fragment twice. That the fragment name was already taken because I'm returning a new fragment possible with __typename on it and adding it to the bottom of this query. I don't really need `gql` and could do it without so it stopped the complaining AND removed a dependency!